### ISSUE TITLE
dev_mode is falsly enabled on nginx/apache setup

### ISSUE THREAD

**Issue by robertharm** [reduxframework/redux-framework#3508]
Hi,

I use worth the read WordPress plugin and found that redux framework is falsely activating dev_mode. After digging through the redux framework code, I found that this is caused by the function `isLocalHost()` in `\class.redux_helpers.php`:

```
public static function isLocalHost() {
                return ( $_SERVER['REMOTE_ADDR'] === '127.0.0.1' || $_SERVER['REMOTE_ADDR'] === 'localhost' ) ? 1 : 0;
            }
```

The issue here is that I am running WordPress on an nginx/apache setup, so Apache is always listening on 127.0.0.1 also on prod. Unfortunately I couldnt find a check on how to enable this function to falsly identify nginx/apache setups as local host installations?

Any help would be appreciated!
thx,

**Comment by kprovance** [comment]
That is by design.  We prefer developres to keep that active to make issue reporting easier, and so our ad network tuns.  If those bother you, you may purchase the ad remove extension.  Your end users will not see those ads and dev tools as long as you keep the dev_mode argument to false.

**Comment by robertharm** [comment]
Maybe I didnt explain the issue detailled enough: it is not the ads that is bothering me but the fact that the function `isLocalHost()` is incorrect for NGINX/apache setups - for our website (and I think for nginx setups in general, although I am not 100% sure), apache runs on 127.0.0.1 and the TLS termination is done by NGINX proxy, which results in redux framework falsly enabling dev_mode even on our production site. 

I tried to find a better solution for identifying local host installations for this setup, but couldnt find any, this is why I thought you would be able to help here. Currently I forked the plugin to always return 0 for `isLocalHost` so that dev_mode does not get enabled on our production site.
